### PR TITLE
#913 Codeforces_Problem_Scraper Added

### LINKED ISSUE #913: Codeforces Scraper
**Describe the solution you'd like**
is a useful website for beginners to practice DSA problems with varying difficulty. This python script will cover the extraction of required problem statements and their related info and storing them in a PDF file.

I am GSSOC21 participant @kaustubhgupta @avinashkranjan

### PR DESCRIPTION
**Pull request by iamakkkhil**
# Description

This python script will let you download Problem Statements from Codeforces and save them as a pdf file. The script uses Selenium Webdriver and fpdf library. Selenium is used with Chrome Webdriver.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [ ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    filename = input('Enter the file name to store Question: ') + '.pdf'
+
+    # Going to the specific URL
+    url = "https://codeforces.com/problemset/problem/" + Pblm_id + "/" + difficulty
+    path = 'image.png'
+    options = webdriver.ChromeOptions()
+
+    # Headless = True for taking a scrolling snapshot
+    options.headless = True
+    driver = webdriver.Chrome(r"chromedriver_win32\chromedriver.exe", options=options)
```
Comment (santushtisharma10) [inline Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py:28]: @iamakkkhil take the input of chrome driver path from user

**Review by santushtisharma10 — CHANGES_REQUESTED** [review]

File: Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import os
+from selenium import webdriver  # Automated webdriver
+from PIL import Image
+from fpdf import FPDF  # For converting images to pdf
+DRIVER_PATH = ''
+
```
Comment (santushtisharma10) [inline Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py:6]:
```suggestion

DRIVER_PATH = input("Enter the chrome driver path : ")


```

**Review by santushtisharma10 — CHANGES_REQUESTED** [review]

File: Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py (hunk 2 above)
Comment (santushtisharma10) [reply Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py:6]: Rest is working fine

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
This is scrapping a single problem. I told you to implement it in the same way as https://github.com/avinashkranjan/Amazing-Python-Scripts/tree/master/Codechef%20Scrapper while assigning the issue. You need to implement like this:

Come up with a script that scraps `n` number of problems from any `category/difficulty` provided by the user and stores all the `n` problems separately in PDFs. The PDF file should contain the problem title, statement, and test cases.

User Input: `n` and `category/difficulty` of the problem
Expected Output: `n` PDF files

File: Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
+import os
+from selenium import webdriver  # Automated webdriver
+from PIL import Image
+from fpdf import FPDF  # For converting images to pdf
+
+DRIVER_PATH = ''
```
Comment (kaustubhgupta) [inline Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py:6]: remove this line as you have shifted this in main function

File: Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+
+def select_difficulty():
+    """
+    This function will let user to choose the difficulty level
+    :return: difficulty_level[]
+    """
+    difficulty_level = []
+    print("\nEnter the Range between 800 to 3500: ")
```
Comment (kaustubhgupta) [inline Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py:15]:
```suggestion
    print("\nEnter the Range of difficulty between 800 to 3500: ")

```

File: Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    level and then by using selenium and chrome webdriver, capturing screenshot of the
+    Codeforces problem using ttypography tag because all the problems of codeforces are
+    stored inside this div tag and saving it in a image.png file.
+    Then saving the image.png as pdf file by using fdf library.
+    """
+
+    path = 'image.png'
+
+    # Creating a Target Output Folder
+    target_folder = './problems_pdf'
```
Comment (kaustubhgupta) [inline Coderforces_Problem_Scrapper/Codeforces_problem_scrapper.py:113]:
```suggestion
    target_folder = './Coderforces_Problem_Scrapper/problems_pdf'

```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Just minor changes, your code works:

**Review by kaustubhgupta — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]

**Pull request merged**